pages/app-layout/controlled-navigation.page.tsx

```diff
@@ -28,92 +29,79 @@ export default function () {
-            id="control-navigation-open"
-            checked={navigationOpen}
-            onChange={e => setUrlParams({ navigationOpen: e.detail.checked })}
-          >
+          <Toggle checked={navigationOpen} onChange={e => setUrlParams({ navigationOpen: e.detail.checked })}>
             Navigation Open
           </Toggle>
-          <Toggle
-            id="control-navigation-hide"
-            checked={navigationHide}
-            onChange={e => setUrlParams({ navigationHide: e.detail.checked })}
-          >
+          <Toggle checked={navigationHide} onChange={e => setUrlParams({ navigationHide: e.detail.checked })}>
             Navigation Hide
           </Toggle>
-          <Toggle
-            id="control-navigation-empty"
-            checked={navigationEmpty}
-            onChange={e => setNavigationEmpty(e.detail.checked)}
-          >
+          <Toggle checked={navigationEmpty} onChange={e => setNavigationEmpty(e.detail.checked)}>
             Navigation Empty
           </Toggle>
         </SpaceBetween>
         <SpaceBetween size="s" direction="horizontal">
-          <Checkbox
-            onChange={e => setUrlParams({ applyNavOpenOverride: e.detail.checked })}
-            checked={applyNavOpenOverride}
-          >
+          <Checkbox onChange={e => setUrlParams({ applyNavOverrides: e.detail.checked })} checked={applyNavOverrides}>
             Apply override
           </Checkbox>
           <Toggle
-            id="override-nav-open"
             checked={!!navOpenOverrideValue}
-            disabled={!applyNavOpenOverride}
+            disabled={!applyNavOverrides}
             onChange={e => setUrlParams({ navOpenOverrideValue: e.detail.checked })}
           >
-            override Navigation Open
+            Navigation Open Override
           </Toggle>
-          <Button
-            id="reset-button"
-            onClick={() => setResetNeeded(true)}
-            disabled={resetNeeded}
-            loading={resetNeeded}
-            data-testid="reset-app-layout"
+          <Toggle
+            checked={!!navHideOverrideValue}
+            disabled={!applyNavOverrides}
+            onChange={e => setUrlParams({ navHideOverrideValue: e.detail.checked })}
           >
-            Reset - Force rerender
-          </Button>
+            Navigation Hide Override
+          </Toggle>
         </SpaceBetween>
+        <Button
+          onClick={() => setResetNeeded(true)}
+          disabled={resetNeeded}
+          loading={resetNeeded}
+          data-testid="reset-app-layout"
+        >
+          Reset - Force rerender
+        </Button>
+
         <Containers />
       </SpaceBetween>
     </>
   );
 
   useEffect(() => {
     if (resetNeeded) {
-      setUrlParams({ applyNavOpenOverride: false, navOpenOverrideValue: null });
+      setUrlParams({ applyNavOverrides: false, navOpenOverrideValue: false, navHideOverrideValue: false });
       setTimeout(() => {
         setResetNeeded(false);
       }, 200);
```

Comment:
Instead of effect, this whole async logic could be inside the reset button onClick handler